DEVELOPMENT.md

```diff
@@ -1,119 +1,238 @@
+* [Acceptance Tests](#acceptance-tests)
+* [Continuous Integration](#continuous-integration)
+* [Releasing a New Version](#releasing-a-new-version)
+* [Miscellaneous Development Notes](#miscellaneous-development-notes)
+
+# ES2015 transition:
+
+Branch `v4.0.0-rc` represents the migration in progress from `Bower`,
+`Grunt` and `requirejs` to a structure following the ES2015 language
+specification, using `npm` to manage dependencies and `webpack` to build
+and bundle the application. The migration strategy was based on the "Treat
+everything like a method" option described by
+[Ben McCormick](https://benmccormick.org/2015/07/06/backbone-and-es6-classes-revisited).
+
+## Dependencies and prerequisites
+
+* [Node](http://nodejs.org/) and [npm](https://www.npmjs.org/)
+* An OpenSearch endpoint capable of receiving search queries and returning
+  search results and related facets.
+
+[Dataset Search Services](https://github.com/nsidc/dataset-search-services/)
+provides the OpenSearch endpoint in the NSIDC environment. See the DSS
+project to learn more about its dependencies, including Solr and Search Solr Tools.
+A local DSS, DSS running in a pre-production environment (e.g., `integration`,
+`qa`, or `staging`), or the production version of DSS may be used as the
+OpenSearch endpoint when running a local instance.  See [Configuration](#configuration), below.
+
+# Git Workflow
 
 Development on this project uses
 [the GitHub Flow](https://guides.github.com/introduction/flow/index.html):
 
 1. Create your feature branch (`git checkout -b my-new-feature`)
 2. Stage your changes (`git add`)
-3. Commit your JSHint-compliant and test-passing changes (just run `grunt`) with a
+3. Commit your ESLint-compliant (`npm run lint`) and test-passing changes
+   (`npm test`) with a
    [good commit message](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
    (`git commit`)
-4. Push to the branch (`git push -u origin my-new-feature`)
-5. [Create a new Pull Request](https://github.com/nsidc/search-interface/compare)
+6. Push to the branch (`git push -u origin my-new-feature`)
+7. [Create a new Pull Request](https://github.com/nsidc/search-interface/compare)
 
-## NSIDC Continuous Integration
+# Installation
 
-NSIDC's project CI utilizes internal configuration found in the `*.yaml` files, `Vagrantfile`
-and `/puppet/*`.   If you are utilizing this project external to NSIDC
-you can safely ignore those files and the remainder of this section.
+1. Install [node.js](http://nodejs.org/) (Check out [NVM](https://github.com/nvm-sh/nvm)
+   if you need to move between versions of node for different projects.)
 
-Most projects just have a file called `vagrant-nsidc.yaml`, which specifies a
-`project` name for the plugin to use. This repository really has two projects,
-`arctic-data-explorer` (formerly ade_search) and `nsidc_search`, configured in
-`vagrant-nsidc.arctic-data-explorer.yaml` and `vagrant-nsidc.nsidc_search.yaml`.
+3. Install node modules:
```

Comment:
Is this the right number?  (It might not matter for Marukup, but maybe this should either be a "2" or a "1" (as numbering everything with "1" is permissible with Markup, I think?)

Reply:
It doesn't matter -- Markup sees it as a request for an element in the same numbered list as the previous item and renders it accordingly (see the list in the Gitflow section). It's the same reason you can use `1` for every element.

Reply:
I can change it though if it bugs you. :-)

Reply:
I'm not that bugged by it, just worried about confusing someone later :)